src/transform.js

```diff
@@ -41,7 +41,16 @@ function _transform (node, transformations, o = {}, property, parent) {
 		}
 		node = transformedNode;
 
-		parents.set(node, parent, {force: true});
+		if (parent) {
+			if (index !== undefined) {
+				parent[property][index] = node;
+			}
+			else {
+				parent[property] = node;
+			}
+			parents.set(node, parent, {force: true});
```

Comment:
1. So here we have both the property and the index, and we have to recompute them, which seems wasteful. What if `parents.set()` allowed providing them in the dictionary, and then it didn't have to calculate anything?
2. For most transformations, simply adjusting the parent pointer of the new node is not enough. Even in the simplest case where we wrap one node with another, the old node has a parent pointer that points to the wrong place. I think we probably need a `setAll()` here, sadly. Unless you can think of a better way?

Reply:
> 1. So here we have both the property and the index, and we have to recompute them, which seems wasteful. What if `parents.set()` allowed providing them in the dictionary, and then it didn't have to calculate anything?

Great idea! I was thinking about proposing this at some point, since usually whenever we use `.set` we have access to property and index

> 2. For most transformations, simply adjusting the parent pointer of the new node is not enough. Even in the simplest case where we wrap one node with another, the old node has a parent pointer that points to the wrong place. I think we probably need a `setAll()` here, sadly. Unless you can think of a better way?

Like a call to `setAll` after the transformation? Although it would add a performance hit, it would significantly simplify the these fixes to transform

Reply:
Yes, on the new node.